core/trino-main/src/main/java/io/trino/sql/planner/optimizations/DeterminePartitionCount.java

```diff
@@ -114,6 +115,11 @@ public PlanNode optimize(
         requireNonNull(types, "types is null");
         requireNonNull(tableStatsProvider, "tableStatsProvider is null");
 
+        // Skip partition count determination if no partitioned remote exchanges exist in the plan anyway
```

Comment:
Javadoc for the plan optimizer is no longer valid (`SELECT count(column_a) FROM table_with_stats_a` case)

Reply:
Will update the javadoc to include a `group by column_b` clause for example 1, and mention it must be a remote partitioned exchange to bring the comment back into line with this change.